content/projects/tdd/string-calculator/_index.md

```diff
@@ -117,6 +117,20 @@ In the first case, the default delimiter is: ";".
 
 In the second case, the default delimiter is: "4".
 
+**Note:** you are not expected to handle integer delimiters where the delimiter and the digit your adding are the same. For example
+
+```
+add("//4\n14244")
+add("//88\n18882")
```

Comment:
So in this case it would mean they would need to specifically code in the error exception to check if the first and last number in the number delimiter is the same as the number before or after it, since
if delimiter = '88'
and string = '18882'
string = string.replace(delimiter, ',')
would give
string =  '1, 82', which is valid.
BUT if string = '188882'
then the same action would give:
string = '1, , 2' 
OR if string = '88882'
then output would be:
string = ', , 2'
both of which are of course easy to demonstrate as invalid. 

So what we actually don't expect their code to do is have the the start of the section to be calculated to start with the same number as the number delimiter or for the numbers on either side to be the whole delimiter, eg 4 next to 4 or 88 next to 88 (since 88 followed by 8 is actually fine)

I'm not exactly sure how best to phrase this, unless we simply expect that they'll need to explicitly check what numbers are around a number delimiter?

Does this make sense?

Reply:
It makes sense but those are implementation details, which we leave up to them. Simply doing `replace(delimiter, ',')` wouldn't be the best thing to do.  

What they would need to do is this; if `delimiter = '88'`, then search `string = '18882'` for an `8` either before or after any instance of the delimiter. If you find one raise an error. It actually won't matter which sequence of `8's` they consider the delimiter, the first two `1[88]82` or the last two `18[88]2`. Either way, they find an `8` before or after and should raise an error. Same thing with the other cases `'188882'`, first two, last two or middle two, either way they'll be an `8` before or after, and an error should be raised.

The function or logic to do this would be a simplified version of the substring problem; here all we want then to do is to make sure that there are no multiple substrings in sequence of the delimiter in the string they need to add.